Add debug only layout assert for empty_cpu

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53617 Marginally improve pytest collection for top-level test files
* #53611 Fix inaccurate dispatch table for fill_
* #53610 Convert a few more checks to raise NotImplementedError
* #53397 Add Meta support for empty_strided
* **#53396 Add debug only layout assert for empty_cpu**
* #53377 Add TORCH_CHECK_NOT_IMPLEMENTED/c10::NotImplementedError; make dispatch use it

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26891806](https://our.internmc.facebook.com/intern/diff/D26891806)

ezyang committed:
Add debug only layout assert for empty_cpu

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

bdhirsh commented on aten/src/ATen/Utils.cpp:
```diff
@@ -31,6 +31,7 @@ Tensor empty_cpu(
     c10::optional<bool> pin_memory_opt,
     c10::optional<c10::MemoryFormat> memory_format_opt) {
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(device_or_default(device_opt).type() == DeviceType::CPU);
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(layout_or_default(layout_opt) == Layout::Strided);
```
Should we add in a corresponding test?

ezyang replied:
no, asserts are supposed to be impossible to reach code, so you can't really test it

ezyang committed:
Update on "Add debug only layout assert for empty_cpu"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26891806](https://our.internmc.facebook.com/intern/diff/D26891806)

[ghstack-poisoned]
